src/auth/AclProxyAuth.cc

```diff
@@ -182,9 +174,21 @@ ACLProxyAuth::matchProxyAuth(ACLChecklist *cl)
             return 0;
         }
     }
+
     /* check to see if we have matched the user-acl before */
-    int result = cacheMatchAcl(&checklist->auth_user_request->user()->proxy_match_cache, checklist);
-    checklist->auth_user_request = NULL;
+    auto &cache = checklist->auth_user_request->user()->proxyAuthAclCache;
+
+    SBuf key(name);
+    if (const auto *v = cache.get(key)) {
+        debugs(28, 4, "cache hit on acl " << key << "=" << *v);
+        checklist->auth_user_request = nullptr;
+        return *v;
+    }
+
+    const auto result = data->match(checklist->auth_user_request->username());
+    debugs(28, 4, "cache miss on acl " << key << ". Adding " << result);
+    cache.add(key, result, (checklist->auth_user_request->user()->expiretime - squid_curtime));
```

Comment:
Where was this cache entry TTL enforced before this PR changes? If this represents a functionality change, please disclose it in the PR.

Reply:
I don't believe this results in any functional change. This cache is stored in the credentials object whose expiretime is being used for the TTL. Both should become invalid at the same clock tick.

Reply:
I assume you are implying that the two caches are always in sync so that when this added entry expires the corresponding credentials object will expire as well and vice versa. Please clarify if that is not the case. If that is the case, is it possible that the credentials object TTL gets updated while this entry TTL stays the same so the two becoming out of sync, which has never happened before (because no second TTL was honored).